molecule/default/tests/test_write_to_attached_storage.py

```
@@ -1,159 +1,107 @@
+    mount_point = '/mnt/test'
+    mount_cmd = ("sudo mount {0}1 {1} &&"
+                 "sudo chmod 777 {1}".format(attach_info.device, mount_point))
+    umount_cmd = "sudo umount {}".format(mount_point)
+    mkfs_cmd = "sudo mkfs.ext4 {}1".format(attach_info.device)
+    test_data = 'Call me Ishmael.'
+    test_file = '{}/test_file.txt'.format(mount_point)
+    parted_mkpart_cmd = ("sudo parted -a opt {} "
+                         "mkpart primary 0% 100%".format(attach_info.device))
+    parted_mklabel_cmd = ("sudo parted {} "
+                          "mklabel msdos".format(attach_info.device))
+    create_mount_point_cmd = ("sudo mkdir -p {}".format(mount_point))
+
+    # Connect to server through SSH
+    ssh = ssh_connect(hostname=small_ubuntu_server.accessIPv4,
+                      username='ubuntu',
+                      retries=15)
+
+    # Partition, format and mount drive
+    assert ssh.exec_command(
+        timeout=60,
+        get_pty=True,
+        command=parted_mklabel_cmd
+    )[1].channel.recv_exit_status() == 0
+
+    assert ssh.exec_command(
+        timeout=60,
+        get_pty=True,
+        command=parted_mkpart_cmd
+    )[1].channel.recv_exit_status() == 0
+
+    assert ssh.exec_command(
+        timeout=60,
+        get_pty=True,
+        command=mkfs_cmd
+    )[1].channel.recv_exit_status() == 0
+
+    assert ssh.exec_command(
+        timeout=60,
+        get_pty=True,
+        command=create_mount_point_cmd
+    )[1].channel.recv_exit_status() == 0
+
+    assert ssh.exec_command(
+        timeout=60,
+        get_pty=True,
+        command=mount_cmd
+    )[1].channel.recv_exit_status() == 0
+
+    # Write test data to mounted drive
+    with ssh.open_sftp().open(test_file, 'w') as f:
+        f.write(test_data)
+
+    # Unmount and remount drive
+    assert ssh.exec_command(
+        timeout=60,
+        get_pty=True,
+        command=umount_cmd
+    )[1].channel.recv_exit_status() == 0
+
+    assert ssh.exec_command(
```

You need to verify that the data is not available after the volume has been unmounted. Otherwise the test may mask a failure to properly attribute the data to the volume. In other words, the data may have been written to the primary storage of the instance without properly verifying the integrity of the external volume.